### PR TITLE
Fix health check query not executed

### LINKED ISSUE #4926: Health check can falsely return OK even if Cosmos is down or inaccessible
**Describe the bug**
in api_app/services/health_checker.py
`container.query_items("SELECT TOP 1 * FROM c")`

This does NOT execute the query.
query_items() returns an async iterable (AsyncItemPaged).
No iteration = no request sent to Cosmos DB.
This means the health check can falsely return OK even if Cosmos is down or inaccessible.

Should be:
`async for _ in container.query_items("SELECT TOP 1 * FROM c", max_item_count=1):
    break`

**Steps to reproduce**

1. Write a test to check the health and see that it does not match

**Azure TRE release version (e.g. v0.14.0 or main):**
main

**Deployed Azure TRE components - click the (i) in the UI:**
latest

### PR DESCRIPTION
**Pull request by JC-wk**
# Resolves #4926

## What is being addressed

The existing code `container.query_items("SELECT TOP 1 * FROM c")` does not actually execute the query so the health check can falsely return OK even if Cosmos is down or inaccessible.

## How is this addressed

- Ensure the query runs
- Add tests
- Update CHANGELOG.md
- Increment version

### CODE REVIEW

**Review by marrobi — COMMENTED** [review]
From Opus 4.8:

Blocking concerns: (1) `create_state_store_status()` still has ambiguous success semantics—`ok` is returned whenever iteration does not raise, including the zero-result path, which can mask real Cosmos connectivity/auth/partition misconfiguration depending on SDK behavior; please make success depend on a definitive successful probe operation (not just “no exception while iterating”) and add an explicit test for empty results.

 (2) The new tests rely on custom async iterator doubles + `MagicMock` that may drift from actual `azure.cosmos.aio.ContainerProxy.query_items` behavior and create false confidence across SDK changes; please tighten fidelity (e.g., `AsyncMock`/shared fixture with realistic async iterable behavior) and assert the exact `query_items` call contract (`"SELECT TOP 1 * FROM c"`, `max_item_count=1`) so regressions are caught.

**Comment by JC-wk** [comment]
> From Opus 4.8:
> 
> Blocking concerns: (1) `create_state_store_status()` still has ambiguous success semantics—`ok` is returned whenever iteration does not raise, including the zero-result path, which can mask real Cosmos connectivity/auth/partition misconfiguration depending on SDK behavior; please make success depend on a definitive successful probe operation (not just “no exception while iterating”) and add an explicit test for empty results.
> 
> (2) The new tests rely on custom async iterator doubles + `MagicMock` that may drift from actual `azure.cosmos.aio.ContainerProxy.query_items` behavior and create false confidence across SDK changes; please tighten fidelity (e.g., `AsyncMock`/shared fixture with realistic async iterable behavior) and assert the exact `query_items` call contract (`"SELECT TOP 1 * FROM c"`, `max_item_count=1`) so regressions are caught.

  In health_checker.py:
  
  • Added an explicit  await container.read()  call before iterating over  container.query_items .
  • Calling  container.read()  acts as a definitive probe of the state store container (validating credentials, endpoints, and partition existence). It will raise a standard database error (           
  ServiceRequestError / CosmosHttpResponseError ) if misconfigured, avoiding ambiguous success status when  query_items  returns an empty result set.
  
  ### 2. High-Fidelity Mocking & Exact Call Contract Assertions
  
  In test_health_checker.py:
  
  • Replaced the custom class-based async iterator double ( AsyncIteratorWithError ) with Python's built-in  AsyncMock  to configure asynchronous iteration. This uses native mocking behavior (         
  __aiter__.return_value = query_results  or  __aiter__.side_effect = query_error ) that matches the Azure SDK iterator interface.
  • Added assertions for the exact  query_items  call contract:  container_mock.query_items.assert_called_once_with("SELECT TOP 1 * FROM c", max_item_count=1) .
  • Added assertions verifying that the new probe operation is invoked:  container_mock.read.assert_called_once() .
  • Added a new explicit test  test_get_state_store_status_empty_results  to verify that when the container is empty, the probe completes successfully and returns  StatusEnum.ok .

File: api_app/services/health_checker.py [hunk 1]
```diff
@@ -19,7 +19,9 @@ async def create_state_store_status() -> Tuple[StatusEnum, str]:
     message = ""
     try:
         container: ContainerProxy = await Database().get_container_proxy(STATE_STORE_RESOURCES_CONTAINER)
-        container.query_items("SELECT TOP 1 * FROM c")
+        await container.read()
+        async for _ in container.query_items("SELECT TOP 1 * FROM c", max_item_count=1):
+            break
```
[the remark this answers is not in the document]
Comment (JC-wk) [reply api_app/services/health_checker.py]: This extra read was added in response to opus feedback to "make success depend on a definitive successful probe operation".
implementation with  await container.read()  +  query_items("SELECT TOP 1 * FROM c")  is the only approach that fully satisfies the Opus 4.8 code review. It ensures we have a definitive   probe that prevents masked failures, while still allowing the health check to succeed on an, empty database.

[the remark this answers is not in the document]
Comment (marrobi) [reply api_app/services/health_checker.py]: Wonder what the RU consumption would be related to this... how often does it run etc.

[the remark this answers is not in the document]
Comment (marrobi) [reply api_app/services/health_checker.py]: Opus:

The extra `await container.read()` isn't needed to satisfy the "definitive probe" requirement — iterating the query (`async for ... break`) already forces a network round-trip to Cosmos, and any failure surfaces as `ServiceRequestError`/`CosmosHttpResponseError` and is caught below. An empty container is still a *successful* probe (the request goes out; the loop body just doesn't run), so `read()` adds no correctness value.

It does add cost: an extra round-trip on every poll and potentially broader RBAC requirements, which could fail the check for permission reasons unrelated to actual availability. I'd drop `read()` and use a minimal projection:

```python
async for _ in container.query_items("SELECT TOP 1 c.id FROM c", max_item_count=1):
    break
```

The unit tests will need updating to match.

[the remark this answers is not in the document]
Comment (JC-wk) [reply api_app/services/health_checker.py]: The costs were only pennies but I'm happy to revert it if Opus has now changed it's mind

[the remark this answers is not in the document]
Comment (marrobi) [reply api_app/services/health_checker.py]: Leave it to you to decide.

**Review by marrobi — APPROVED** [review]

File: api_app/services/health_checker.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JC-wk) [reply api_app/services/health_checker.py]: reverted
